src/core/mapping/base_mapper.cc

```diff
@@ -928,54 +928,68 @@ void BaseMapper::select_task_sources(const MapperContext ctx,
                                      const SelectTaskSrcInput& input,
                                      SelectTaskSrcOutput& output)
 {
-  legate_select_sources(ctx, input.target, input.source_instances, output.chosen_ranking);
+  legate_select_sources(
+    ctx, input.target, input.source_instances, input.collective_views, output.chosen_ranking);
+}
+
+void add_instance_to_band_ranking(const PhysicalInstance& instance,
+                                  const Legion::AddressSpace& local_node,
+                                  std::map<Memory, uint32_t /*bandwidth*/>& source_memories,
+                                  std::vector<std::pair<PhysicalInstance, uint32_t>>& band_ranking,
+                                  std::vector<MemoryMemoryAffinity>& affinity,
+                                  const Memory& destination_memory,
+                                  const Legion::Machine& machine)
+{
+  Memory location = instance.get_location();
+  auto finder     = source_memories.find(location);
+  if (finder == source_memories.end()) {
+    affinity.clear();
```

Comment:
I think we can just declare `affinity` as a temporary here, instead of passing it and clearing it.